ddtrace/appsec/_asm_request_context.py

```diff
@@ -251,12 +267,13 @@ def finalize_asm_env(env: ASM_Environment) -> None:
             try:
                 if info.errors:
                     root_span.set_tag_str(APPSEC.EVENT_RULE_ERRORS, info.errors)
-                    log.debug("appsec.asm_context.debug::finalize_asm_env::waf_errors::%s", info.errors)
+                    extra = {"product": "appsec", "more_info": info.errors, "stack_limit": 4}
+                    logger.debug("asm_context::finalize_asm_env::waf_errors", extra=extra, stack_info=True)
```

Comment:
Why not use a constant here for the message?

Reply:
because it's only used there. I defined constants if the tag is used at least twice.